worker/client/client.go

```diff
@@ -194,6 +194,7 @@ func (c *Client) UploadMultipartUploadPart(ctx context.Context, r io.Reader, buc
 		panic(err)
 	}
 	req.SetBasicAuth("", c.c.WithContext(ctx).Password)
+	req.Header.Add("X-Content-Length", fmt.Sprintf("%d", contentLength))
```

Comment:
Try setting `req.ContentLength = contentLength` instead. That way the `Content-Length` header will be set.
But `contentLength` should be part of `opts` and only be set if `opts.ContentLength != 0`.

It would also help us a lot if you did the same for `PutObject` in `backend.go` since that also seems to have a `size` that is currently not used.

Reply:
If you want to go the extra mile you can extend the client methods to try and cast `r.(io.Seeker)` iff `opts.ContentLength = 0` and then seed to the end and back to the beginning to determine the content length of the input. That would make sure that calling the client methods with an `os.File` or `bytes.Reader` will set the content length implicitly.

Reply:
Oh sry I misunderstood your question. We still need to set the content length. Just not from the input but from the offset of the seeker.

Please also make sure to handle all errors